### ISSUE TITLE
Add fallback execution support

### ISSUE THREAD

**Issue by jhalterman** [jhalterman/failsafe#17]
Add support for fallback actions which are executed when a failure occurs and cannot be retried:

```
Failsafe...recoverWith(executor).get(executor);
```

Or possibly:

```
Failsafe...tryRun(foo).elseRun(bar);
Failsafe...tryRun(foo, executor).elseRun(bar, executor);
Failsafe...tryGet(this::connect).elseGet(this:reconnect);
```

**Comment by jhalterman** [comment]
Implemented via 22934ea5eb863c3b0e0d5deb5c3c8c8cf2570d09.

**Comment by jhalterman** [comment]
Docs https://github.com/jhalterman/failsafe#fallbacks
